### PR TITLE
Add component minimap and include into map debug tab

### PR DESCRIPTION
**Pull request by vitorsilverio**

### CODE REVIEW

**Comment by vitorsilverio** [comment]
@hanzi I extracted the function to extract all tiles to map module and created a cache for map images

**Comment by vitorsilverio** [comment]
@hanzi, sry my mistake, i used the the wrong tuple, the idea of the cache is keep map for map group/number

File: modules/gui/debug_tabs.py [hunk 1]
```diff
@@ -193,6 +194,69 @@ def _handle_action(self, callback: callable) -> None:
...
+                tile_color = MapViewer.NORMAL
+                if bool(tile_data.collision):
+                    tile_color = MapViewer.COLLISION
+                if tile_data.has_encounters:
+                    tile_color = MapViewer.ENCOUNTERS
+                if "Jump" in tile_data.tile_type:
+                    tile_color = MapViewer.JUMP
+                if tile_data.is_surfable:
+                    tile_color = MapViewer.WATER
+                image_draw.rectangle(xy = [
```
Comment (40Cakes) [inline modules/gui/debug_tabs.py:249]:
```
Expected type 'tuple[float, float, float, float] | tuple[tuple[float, float], tuple[float, float]]', got 'list[tuple[int, int]]' instead
```

**Comment by 40Cakes** [comment]
> @40Cakes i fixed the comments

Thanks mate, great work on this btw, looks fantastic.

**Pull request merged**